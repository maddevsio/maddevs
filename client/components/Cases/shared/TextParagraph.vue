<template>
  <p
    :style="textColor"
    class="case_paragraph"
  >
    <slot />
  </p>
</template>

<script>
export default {
  name: 'TextParagraph',
  props: {
    color: {
      type: String,
      default: '',
    },
  },

  computed: {
    textColor() {
      if (this.color) {
        return `color: ${this.color}`
      }
      return null
    },
  },
}
</script>

<style lang="scss" scoped>
@import '../../../assets/styles/cases/_mixins';

.case {
  &_paragraph {
    @include default_text($text-color--black-oil, 17px, 166%, -0.035em, normal);

    @media screen and (max-width: 768px) {
      @include default_text($text-color--black-oil, 16px, 150%, -0.02em, normal);
    }

    &--bold {
      font-weight: 700;
    }
  }
}
</style>
