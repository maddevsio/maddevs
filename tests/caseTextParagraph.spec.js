import {
  mount
} from '@vue/test-utils';
import TextParagraph from '@/components/Cases/TextParagraph';

describe('Text paragraph', () => {
  let wrapper;

  beforeEach(() => {
    wrapper = mount(TextParagraph, {
      propsData: {
        color: '#000000'
      },
      slots: {
        default: 'Main Content'
      }
    });
  });

  // ------ IMPORTANT ----- //
  test('is a Vue instance', () => {
    expect(wrapper.exists()).toBeTruthy();
  });
  
  test('renders correctly', () => {
    expect(wrapper.element).toMatchSnapshot();
  });
  // --------------------- //

  test('should have correct data in slot', () => {
    expect(wrapper.html()).toContain('Main Content');
  });

  test('should have correct style', () => {
    const paragraph = wrapper.find('.case_paragraph');
    expect(paragraph.element.style.getPropertyValue('color')).toEqual('rgb(0, 0, 0)');
  });
});
