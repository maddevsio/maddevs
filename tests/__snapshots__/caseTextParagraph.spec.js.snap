// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Text paragraph renders correctly 1`] = `
<p
  class="case_paragraph"
  style="color: rgb(0, 0, 0);"
>
  Main Content
</p>
`;
